File: src/main/java/jp/co/sunarch/apps/idprovider/model/oauth2/implicit/ImplicitFlow.java
package jp.co.sunarch.apps.idprovider.model.oauth2.implicit;

import jp.co.sunarch.apps.idprovider.model.oauth2.AuthorizationEndpoint;
import jp.co.sunarch.apps.idprovider.model.oauth2.EndpointRequestFactory;
import jp.co.sunarch.apps.idprovider.model.oauth2.AccessTokenResponse;

/**
 * @author takeshi
 */
public class ImplicitFlow implements AuthorizationEndpoint<ImplicitAuthorizationRequest, AccessTokenResponse> {

  @Override
  public EndpointRequestFactory<ImplicitAuthorizationRequest> authorizationRequestFactory() {
    // TODO
    return null;
  }

  @Override
  public AccessTokenResponse authorize(ImplicitAuthorizationRequest request) {
    // TODO
    return null;
  }

}

File: src/main/kotlin/jp/co/sunarch/apps/idprovider/model/oauth2/implicit/ImplicitFlow.kt
package jp.co.sunarch.apps.idprovider.model.oauth2.implicit

import jp.co.sunarch.apps.idprovider.model.oauth2.AuthorizationEndpoint
import jp.co.sunarch.apps.idprovider.model.oauth2.EndpointRequestFactory
import jp.co.sunarch.apps.idprovider.model.oauth2.AccessTokenResponse

/**
 * @author takeshi
 */
class ImplicitFlow : AuthorizationEndpoint<ImplicitAuthorizationRequest, AccessTokenResponse> {

  override fun authorizationRequestFactory(): EndpointRequestFactory<ImplicitAuthorizationRequest> {
    TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
  }

  override fun authorize(request: ImplicitAuthorizationRequest): AccessTokenResponse {
    TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
  }

}
